src/elements/content-sidebar/Sidebar.js

```diff
@@ -111,8 +125,31 @@ class Sidebar extends React.Component<Props, State> {
             this.setForcedByLocation();
             this.setState({ isDirty: true });
         }
+
+        this.handleDocgenTemplateOnUpdate(prevProps);
     }
 
+    handleDocgenTemplateOnUpdate = (prevProps: Props) => {
+        const { history, location, file, api, metadataSidebarProps, docGenSidebarProps } = this.props;
+        const { file: prevFile, docGenSidebarProps: prevDocGenSidebarProps }: Props = prevProps;
+        // need to re-check if file is a docgen-template on file change
+        if (file.id !== prevFile.id && docGenSidebarProps?.enabled && docGenSidebarProps?.checkDocGenTemplate) {
+            docGenSidebarProps.checkDocGenTemplate(api, file, metadataSidebarProps.isFeatureEnabled);
```

Comment:
Should we have an await here? Is checkDocGenTemplate async?

Reply:
no that's not necessary, checkDocGenTemplate will reset the isDocgenTemplate prop which will be responsible for deciding whether to display or not display the sidebar and whether to navigate to the docgen tab